File: hw/bonilla/u1/hw04MultiplicationTables/MultiplicationTables.py
def show_table(number):
    print(f"\n{'-'*31}")
    print(f"Multiplication Table of {number}".center(31))
    print(f"{'-'*31}")
    for i in range(1, 13):
        print(f"{number:>2} x {i:>2} = {number*i:>3}")
    print(f"{'-'*31}\n")

def ask_for_table():
    """Asks for a valid table number (1–12) and returns it."""
    while True:
        try:
            choice = int(input("Enter the number of the table you want to see: "))
            if 1 <= choice <= 12:
                return choice
            else:
                print("\nError: Please choose a number between 1 and 12.\n")
        except ValueError:
            print("\nError: You must enter a valid number (no letters or symbols).\n")

def main():
    print("------ Welcome to the Multiplication Table Program ------")
    print("You can choose a table from 1 to 12.\n")

    while True:
        number = ask_for_table()
        show_table(number)

        again = input("Do you want to see another table? (yes/no): ").strip().lower()
        if again not in ("yes", "y"):
            print("\nThank you for using the program.\n")
            break
        print() 

if __name__ == "__main__":
    main()

File: hw/bonilla/u1/hw04MultiplicationTables/MultiplicationTables.js
const readline = require("readline");

function showTable(number) {
  console.log("\n" + "-".repeat(31));
  console.log(`   Multiplication Table of ${number}`.padStart(22 + Math.floor(number / 10)));
  console.log("-".repeat(31));
  for (let i = 1; i <= 12; i++) {
    console.log(`${String(number).padStart(2)} x ${String(i).padStart(2)} = ${String(number * i).padStart(3)}`);
  }
  console.log("-".repeat(31) + "\n");
}

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function askForNumber(callback) {
  rl.question("Enter the number of the table you want to see: ", (input) => {
    const number = parseInt(input);

    if (isNaN(number)) {
      console.log("\nError: You must enter a valid number (no letters or symbols).\n");
      return askForNumber(callback);
    }

    if (number < 1 || number > 12) {
      console.log("\nError: Please choose a number between 1 and 12.\n");
      return askForNumber(callback);
    }

    showTable(number);
    callback(); 
  });
}

function askToContinue() {
  rl.question("Do you want to see another table? (yes/no): ", (answer) => {
    const response = answer.trim().toLowerCase();
    if (response === "yes" || response === "y") {
      console.log();
      askForNumber(askToContinue);
    } else {
      console.log("\nThank you for using the program.\n");
      rl.close();
    }
  });
}

function main() {
  console.log("----- Welcome to the Multiplication Table Program -----");
  console.log("You can choose a table from 1 to 12.\n");
  askForNumber(askToContinue);
}

main();